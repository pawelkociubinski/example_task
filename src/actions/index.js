/* @flow */
import type { OrderBy, SortByProperty } from '../types/models';
import type {
  ChangePageEvent, PreviousPageEvent, NextPageEvent,
  GithubTokenRequestedEvent, SortItemsPerPageEvent,
  SearchTermRequestedEvent, SortColumnEvent, LogoutEvent,
} from '../types/actions';

import * as actions from '../constants/actions';

export const logout = (): LogoutEvent => ({
  type: actions.LOGOUT,
});

export const searchTerm = (searchedTerm: string): SearchTermRequestedEvent => ({
  type: actions.SEARCH_TERM_REQUESTED,
  payload: { searchedTerm },
});

export const sortByColumn = (sortByProperty: SortByProperty, orderBy: OrderBy): SortColumnEvent => ({
  type: actions.SORT_COLUMN_BY_ORDER,
  payload: { sortByProperty, orderBy },
});

export const changePage = (page: number): ChangePageEvent => ({
  type: actions.CHANGE_PAGE,
  payload: { page },
});

export const previousPage = (): PreviousPageEvent => ({
  type: actions.PREVIOUS_PAGE,
});

export const nextPage = (): NextPageEvent => ({
  type: actions.NEXT_PAGE,
});

export const getGithubToken = (code: string): GithubTokenRequestedEvent => ({
  type: actions.GITHUB_TOKEN_REQUESTED,
  payload: { code },
});

export const sortItemsPerPage = (itemsPerPage: number): SortItemsPerPageEvent => ({
  type: actions.SORT_ITEMS_PER_PAGE,
  payload: { itemsPerPage },
});
