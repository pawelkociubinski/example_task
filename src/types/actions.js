/* @flow */

import type { OrderBy } from 'types/models';

export type LogoutEvent = {|
  type: string,
|}

export type SearchTermRequestedEvent = {|
  type: string,
  payload: {|
    searchedTerm: string
  |}
|};

export type PreviousPageEvent = {|
  type: string,
|}

export type NextPageEvent = {|
  type: string,
|}

export type ChangePageEvent = {|
  type: string,
  payload: {|
    page: number,
  |}
|}

export type SortColumnEvent = {|
  type: string,
  payload: {|
    sortByProperty: string,
    orderBy: OrderBy,
  |}
|}

export type GithubTokenRequestedEvent = {|
  type: string,
  payload: {|
    code: string,
  |}
|}

export type SortItemsPerPageEvent = {|
  type: string,
  payload: {|
    itemsPerPage: number,
  |}
|}

export type Action =
  ChangePageEvent
  | GithubTokenRequestedEvent
  | LogoutEvent
  | NextPageEvent
  | NextPageEvent
  | PreviousPageEvent
  | SearchTermRequestedEvent
  | SortColumnEvent
  | SortItemsPerPageEvent

export type SearchTermEvent = SearchTermRequestedEvent;
