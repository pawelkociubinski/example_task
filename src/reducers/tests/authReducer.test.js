
import reducer from '../authReducer';
import * as actions from '../../constants/actions';

describe('authReducer', () => {
  it('should return initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle FETCH_GITHUB_TOKEN_STARTED', () => {
    const action = {
      type: actions.FETCH_GITHUB_TOKEN_STARTED,
    };

    expect(reducer(undefined, action)).toMatchSnapshot();
  });

  it('should handle FETCH_GITHUB_TOKEN_SUCCEEDED', () => {
    const action = {
      type: actions.FETCH_GITHUB_TOKEN_SUCCEEDED,
      payload: {
        token: 'token',
      },
    };

    expect(reducer(undefined, action)).toMatchSnapshot();
  });

  it('should handle FETCH_GITHUB_TOKEN_FINISHED', () => {
    const action = {
      type: actions.FETCH_GITHUB_TOKEN_FINISHED,
    };

    expect(reducer(undefined, action)).toMatchSnapshot();
  });

  it('should handle FETCH_GITHUB_TOKEN_FAILED', () => {
    const action = {
      type: actions.FETCH_GITHUB_TOKEN_FAILED,
      payload: {
        error: true,
      },
    };

    expect(reducer(undefined, action)).toMatchSnapshot();
  });
});
