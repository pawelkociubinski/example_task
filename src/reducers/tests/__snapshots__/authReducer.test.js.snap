// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authReducer should handle FETCH_GITHUB_TOKEN_FAILED 1`] = `
Object {
  "error": true,
  "inProgress": false,
  "token": null,
}
`;

exports[`authReducer should handle FETCH_GITHUB_TOKEN_FINISHED 1`] = `
Object {
  "error": false,
  "inProgress": false,
  "token": null,
}
`;

exports[`authReducer should handle FETCH_GITHUB_TOKEN_STARTED 1`] = `
Object {
  "error": false,
  "inProgress": true,
  "token": null,
}
`;

exports[`authReducer should handle FETCH_GITHUB_TOKEN_SUCCEEDED 1`] = `
Object {
  "error": false,
  "inProgress": false,
  "token": "token",
}
`;

exports[`authReducer should return initial state 1`] = `
Object {
  "error": false,
  "inProgress": false,
  "token": null,
}
`;
